jung-samples/src/main/java/edu/uci/ics/jung/samples/ClusteringDemo.java

```diff
@@ -133,7 +133,13 @@ public Number get() {
     //specify the Fruchterman-Rheingold layout algorithm
     LayoutAlgorithm<Number, Point2D> algorithm = new FRLayoutAlgorithm<>(POINT_MODEL);
```

Comment:
Is there a way for a `LayoutAlgorithm` or a `LayoutModel` to be able to infer the PointModel that they need based on the point type?  I haven't thought this through clearly so there may be some obvious reason why that doesn't work (or is a bad idea).

Reply:
That would be very cool. Let's think about it.

Reply:
I guess you could write a LayoutAlgorithm with only x, y and write all the math yourself, and give up any optimizations provided by a library class (like Point2D) that _anyone_ writing a visualization system should have access to. I prefer using code that works and is suspected to be fast over writing my own, especially for simple things.

Reply:
I'll make it so the LayoutAlgorithm visit call to the LayoutModel sets the PointModel in the LayoutAlgorithm. It will be in the next (massive) push to this branch that includes all my pick fixing